lib/puppet/functions/foreman/enc.rb

```diff
@@ -0,0 +1,108 @@
+Puppet::Functions.create_function(:'foreman::enc') do
+  dispatch :foreman_enc do
+    # Copied from Stdlib::HTTPUrl
+    param 'Struct[{url=>Pattern[/(?i:^https?:\/\/)/]}]', :options
+    param 'Puppet::LookupContext', :context
+  end
+
+  argument_mismatch :missing_path do
+    param 'Hash', :options
+    param 'Puppet::LookupContext', :context
+  end
+
+  def parse_error(body)
+    begin
+      require 'json'
+      content = JSON.parse(body)
+    rescue
+      content = body
+    end
+
+    begin
+      content['error']['message']
+    rescue
+      "Check Foreman's /var/log/foreman/production.log for more information."
```

Comment:
I wonder if it's a good idea to hard code the log path here, as it might differ?

Reply:
Updated